isis/src/tgo/apps/tgocassisunstitch/tgocassisunstitch.cpp

```diff
@@ -152,16 +156,29 @@ void IsisMain() {
   for (int i = 0; i < g_outputCubes.size(); i++) {
     progress.CheckStatus();
     for (int j = 0; j < inputLabel->findObject("IsisCube").groups(); j++) {
-      g_outputCubes[i]->putGroup(inputLabel->findObject("IsisCube").group(j));
+      PvlGroup group = inputLabel->findObject("IsisCube").group(j);
+
+      // The stitched frame has ArchiveRED, ArchiveNIR, ArchivePAN, and ArchiveBLU.
+      // We won't add the archive group unless
+      if ( group.name().contains("Archive") &&
+           group.name() != "Archive" + g_frameletInfoList[i].m_filterName ) {
+             continue;
+           }
+
+      g_outputCubes[i]->putGroup(group);
```

Comment:
This is going to make an unstitch NIR cube's archive group called ArchiveNIR. It should be just Archive. You'll need to change the group name before you add it to the unstitched cube.

Reply:
Nevermind, just read the next part